File: scripts/convert-to-yaml.py
#!/usr/bin/env python3
"""
Convert Markdown format knowledge base files to YAML format
"""
import re
import os

def convert_markdown_to_yaml(content):
    """Convert Markdown format to YAML format"""
    # Split by sections (## headings)
    # Each section starts with ## Topic, then ### Subtopic, then metadata
    sections = re.split(r'^##\s+', content, flags=re.MULTILINE)
    # Filter out empty sections and keep only those with content
    sections = [s for s in sections if s.strip()]
    
    # Re-add the ## prefix for processing (skip first empty split if file starts with ##)
    if sections and not sections[0].strip().startswith('###'):
        sections = sections[1:]  # Remove first empty element if file starts with ##
    sections = [f'## {s}' for s in sections if s.strip()]
    
    yaml_sections = []
    
    for section in sections:
        # Extract metadata - try both formats
        persona_match = re.search(r'\*\*Persona:\*\*\s*([^\r\n]+)', section)
        topic_match = re.search(r'\*\*Topic:\*\*\s*([^\r\n]+)', section)
        subtopic_match = re.search(r'\*\*Subtopic:\*\*\s*([^\r\n]+)', section)
        
        # If metadata not found in ** format, try extracting from headings
        if not persona_match or not topic_match or not subtopic_match:
            topic_line = re.search(r'^##\s+(.+?)(?:\s*$|\n)', section, re.MULTILINE)
            subtopic_line = re.search(r'^###\s+(.+?)(?:\s*$|\n)', section, re.MULTILINE)
            if topic_line and subtopic_line:
                topic = topic_line.group(1).strip()
                subtopic = subtopic_line.group(1).strip()
                # Try to find persona
                persona_in_section = re.search(r'\*\*Persona:\*\*\s*([^\r\n]+)', section)
                if persona_in_section:
                    persona = persona_in_section.group(1).strip()
                else:
                    # Default persona if not found
                    persona = "menopause"
            else:
                print(f"Warning: Skipping section - cannot extract topic/subtopic")
                continue
        else:
            persona = persona_match.group(1).strip()
            topic = topic_match.group(1).strip()
            subtopic = subtopic_match.group(1).strip()
        
        # Extract content - stop at next ### or --- separator
        content_match = re.search(r'###\s+\*\*Content\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        content_text = content_match.group(1).strip() if content_match else ''
        # Clean up content - remove trailing separators
        content_text = re.sub(r'\n*---\s*\n*$', '', content_text, flags=re.MULTILINE).strip()
        
        # Extract action tips
        action_tips_match = re.search(r'###\s+\*\*Action Tips?\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        action_tips = []
        if action_tips_match:
            tips_text = action_tips_match.group(1)
            for line in tips_text.split('\n'):
                line = line.strip()
                if line and line.startswith('-') and not line.startswith('- [') and line not in ['---', '--']:
                    tip = re.sub(r'^-\s*', '', line).strip()
                    # Remove bold markers if present
                    tip = re.sub(r'\*\*([^*]+)\*\*', r'\1', tip)
                    if tip:
                        action_tips.append(tip)
        
        # Extract motivation nudge
        motivation_match = re.search(r'###\s+\*\*Motivation Nudge\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        motivation_nudge = ''
        if motivation_match:
            motivation_nudge = motivation_match.group(1).strip()
            motivation_nudge = re.sub(r'\n*---\s*\n*$', '', motivation_nudge, flags=re.MULTILINE).strip()
            # Remove bold markers
            motivation_nudge = re.sub(r'\*\*([^*]+)\*\*', r'\1', motivation_nudge)
        
        # Extract habit strategy
        habit_strategy_match = re.search(r'###\s+\*\*Habit Strategy[^\n]*\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        habit_strategy = None
        if habit_strategy_match:
            strategy_text = habit_strategy_match.group(1)
            principle_match = re.search(r'\*\*(?:Strategy|Principle):\*\*\s*([^\n]+)', strategy_text)
            explanation_match = re.search(r'\*\*Explanation:\*\*\s*([^\n]+)', strategy_text)
            example_match = re.search(r'\*\*Example:\*\*\s*([^\n]+)', strategy_text)
            tip_match = re.search(r'\*\*Habit Tip:\*\*\s*([^\n]+)', strategy_text)
            
            if principle_match or explanation_match or example_match or tip_match:
                habit_strategy = {
                    'principle': principle_match.group(1).strip() if principle_match else '',
                    'explanation': explanation_match.group(1).strip() if explanation_match else '',
                    'example': example_match.group(1).strip() if example_match else '',
                    'habit_tip': tip_match.group(1).strip() if tip_match else '',
                }
        
        # Extract follow-up question(s)
        follow_up_match = re.search(r'###\s+\*\*Follow-Up Questions?\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        follow_up_question = ''
        if follow_up_match:
            follow_up_text = follow_up_match.group(1).strip()
            follow_up_text = re.sub(r'\n*---\s*\n*$', '', follow_up_text, flags=re.MULTILINE).strip()
            # Handle multiple questions - take first one
            if follow_up_text.startswith('-'):
                questions = [q.strip() for q in follow_up_text.split('\n') if q.strip().startswith('-') and q.strip() != '---']
                if questions:
                    follow_up_question = questions[0].replace('-', '').strip()
                    # Remove bold markers
                    follow_up_question = re.sub(r'\*\*([^*]+)\*\*', r'\1', follow_up_question)
            else:
                follow_up_question = follow_up_text
                # Remove bold markers
                follow_up_question = re.sub(r'\*\*([^*]+)\*\*', r'\1', follow_up_question)
        
        # Extract intent patterns
        intent_patterns_match = re.search(r'###\s+\*\*Intent Patterns?\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        intent_patterns = []
        if intent_patterns_match:
            patterns_text = intent_patterns_match.group(1)
            for line in patterns_text.split('\n'):
                line = line.strip()
                # Skip headers like "PRIMARY INTENTS"
                if line and line.startswith('-') and not line.startswith('- [') and not re.match(r'^-\s*[A-Z\s]+INTENTS?', line):
                    pattern = re.sub(r'^-\s*', '', line).strip()
                    # Remove trailing notes in brackets
                    pattern = re.sub(r'\s*\[\+.*?\]\s*$', '', pattern).strip()
                    if pattern and len(pattern) >= 3:
                        intent_patterns.append(pattern)
        
        # Extract keywords
        keywords_match = re.search(r'###\s+\*\*Keywords?\*\*\s*\n(.*?)(?=\n###\s+\*\*|\n---\s*$|$)', section, re.DOTALL | re.MULTILINE)
        keywords = []
        if keywords_match:
            keywords_text = keywords_match.group(1)
            for line in keywords_text.split('\n'):
                line = line.strip()
                # Skip category headers, shared/route notes, and empty lines
                if line and line.startswith('-') and not re.search(r'\[shared|\[→', line) and line not in ['---', '--']:
                    keyword = re.sub(r'^-\s*', '', line).strip()
                    # Remove shared/route notes
                    keyword = re.sub(r'\s*\[shared.*?\]\s*$', '', keyword).strip()
                    keyword = re.sub(r'\s*\[→.*?\]\s*$', '', keyword).strip()
                    # Skip category headers like "**Core Nutrition Concepts**"
                    if keyword and not keyword.startswith('**') and not keyword.endswith('**'):
                        keywords.append(keyword)
        
        # Build YAML section
        yaml_section = '---\n'
        yaml_section += f'persona: "{persona}"\n'
        yaml_section += f'topic: "{topic}"\n'
        yaml_section += f'subtopic: "{subtopic}"\n\n'
        
        yaml_section += 'content_text: |\n'
        if content_text:
            for line in content_text.split('\n'):
                yaml_section += f'  {line}\n'
        else:
            yaml_section += '\n'
        
        if action_tips:
            yaml_section += '\naction_tips:\n'
            for tip in action_tips:
                tip_escaped = tip.replace('"', '\\"').replace('\n', ' ')
                yaml_section += f'  - "{tip_escaped}"\n'
        
        if motivation_nudge:
            motivation_escaped = motivation_nudge.replace('"', '\\"').replace('\n', ' ')
            yaml_section += f'\nmotivation_nudge: "{motivation_escaped}"\n'
        
        if habit_strategy and (habit_strategy.get('principle') or habit_strategy.get('explanation')):
            yaml_section += '\nhabit_strategy:\n'
            if habit_strategy.get('principle'):
                principle_escaped = habit_strategy['principle'].replace('"', '\\"')
                yaml_section += f'  principle: "{principle_escaped}"\n'
            if habit_strategy.get('explanation'):
                explanation_escaped = habit_strategy['explanation'].replace('"', '\\"')
                yaml_section += f'  explanation: "{explanation_escaped}"\n'
            if habit_strategy.get('example'):
                example_escaped = habit_strategy['example'].replace('"', '\\"')
                yaml_section += f'  example: "{example_escaped}"\n'
            if habit_strategy.get('habit_tip'):
                tip_escaped = habit_strategy['habit_tip'].replace('"', '\\"')
                yaml_section += f'  habit_tip: "{tip_escaped}"\n'
        
        if follow_up_question:
            follow_up_escaped = follow_up_question.replace('"', '\\"').replace('\n', ' ')
            yaml_section += f'\nfollow_up_question: "{follow_up_escaped}"\n'
        
        if intent_patterns:
            yaml_section += '\nintent_patterns:\n'
            for pattern in intent_patterns:
                pattern_escaped = pattern.replace('"', '\\"')
                yaml_section += f'  - "{pattern_escaped}"\n'
        
        if keywords:
            yaml_section += '\nkeywords:\n'
            for keyword in keywords:
                keyword_escaped = keyword.replace('"', '\\"')
                yaml_section += f'  - "{keyword_escaped}"\n'
        
        yaml_section += '---\n\n'
        yaml_sections.append(yaml_section)
    
    return ''.join(yaml_sections)

# Convert files
kb_dir = os.path.join(os.path.dirname(__file__), '..', 'knowledge-base')
files_to_convert = [
    'Metabolic Changes & Weight Gain.md',
    'Sleep Disturbances.md',
]

for filename in files_to_convert:
    file_path = os.path.join(kb_dir, filename)
    if os.path.exists(file_path):
        print(f'Converting {filename}...')
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        yaml_content = convert_markdown_to_yaml(content)
        if yaml_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(yaml_content)
            print(f'Converted {filename} - {len(yaml_content)} characters written')
        else:
            print(f'Warning: No content generated for {filename}')
    else:
        print(f'Warning: {filename} not found')

print('Conversion complete!')

File: scripts/convert-to-yaml.ts
/**
 * Convert Markdown format knowledge base files to YAML format
 * Reuses parsing logic from ingest-documents.ts for accuracy
 */

import fs from 'fs';
import path from 'path';

// Copy the parsing functions from ingest-documents.ts
function parseMarkdownSections(content: string): string[] {
  const sections: string[] = [];
  const lines = content.split(/\r?\n/);
  let currentSection: string[] = [];
  let lastTopicHeading: string | null = null;
  
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    const trimmed = line.trim();
    
    if (line.match(/^\s*##\s+/)) {
      lastTopicHeading = line;
      continue;
    }
    
    const isSubtopicHeading = trimmed.match(/^###\s+/);
    
    if (isSubtopicHeading) {
      let isFollowedByMetadata = false;
      for (let j = i + 1; j < Math.min(i + 6, lines.length); j++) {
        const nextLine = lines[j].trim();
        if (nextLine.includes('**Persona:**') || 
            nextLine.includes('**Topic:**') || 
            nextLine.includes('**Subtopic:**')) {
          isFollowedByMetadata = true;
          break;
        }
        if (nextLine.match(/^#{1,3}\s+/)) {
          break;
        }
      }
      
      if (isFollowedByMetadata && currentSection.length > 0) {
        sections.push(currentSection.join('\n'));
        currentSection = [];
        if (lastTopicHeading) {
          currentSection.push(lastTopicHeading);
        }
      }
    }
    
    currentSection.push(line);
  }
  
  if (currentSection.length > 0) {
    sections.push(currentSection.join('\n'));
  }
  
  return sections;
}

function extractMarkdownMetadata(section: string): { persona: string; topic: string; subtopic: string } | null {
  const personaMatch = section.match(/\*\*Persona:\*\*\s*([^\r\n]+?)(?:\s*[\r\n]|$)/);
  const topicMatch = section.match(/\*\*Topic:\*\*\s*([^\r\n]+?)(?:\s*[\r\n]|$)/);
  const subtopicMatch = section.match(/\*\*Subtopic:\*\*\s*([^\r\n]+?)(?:\s*[\r\n]|$)/);
  
  if (!personaMatch || !topicMatch || !subtopicMatch) {
    return null;
  }
  
  return {
    persona: personaMatch[1].trim(),
    topic: topicMatch[1].trim(),
    subtopic: subtopicMatch[1].trim(),
  };
}

function extractContentText(section: string): string {
  const contentMatch = section.match(/###\s*\*\*Content\*\*\s*\n([\s\S]*?)(?=###\s*\*\*(?:Action|Motivation|Habit|Follow-Up|Intent|Keywords)|---\s*$|$)/i);
  let content = contentMatch ? contentMatch[1].trim() : '';
  // Remove stray separators from content
  content = content.replace(/^---\s*$/gm, '').trim();
  return content;
}

function extractActionTips(section: string): string[] {
  const tips: string[] = [];
  const actionTipsMatch = section.match(/###\s*\*\*Action Tips?\*\*\s*\n([\s\S]*?)(?=###\s*\*\*(?:Content|Motivation|Habit|Follow-Up|Intent|Keywords)|---\s*$|$)/i);
  
  if (actionTipsMatch) {
    const tipsText = actionTipsMatch[1];
    const tipLines = tipsText.split('\n');
    for (const line of tipLines) {
      const trimmed = line.trim();
      if (trimmed.startsWith('-') || trimmed.startsWith('•')) {
        let tip = trimmed.replace(/^[-•]\s*/, '').trim();
        tip = tip.replace(/\*\*([^*]+):\*\*\s*/g, '$1: ');
        // Skip empty tips or separators
        if (tip && tip !== '--' && tip !== '---') {
          tips.push(tip);
        }
      }
    }
  }
  
  return tips;
}

function extractMotivationNudge(section: string): string {
  const motivationMatch = section.match(/###\s*\*\*Motivation Nudge\*\*\s*\n([\s\S]*?)(?=###\s*\*\*(?:Content|Action|Habit|Follow-Up|Intent|Keywords)|---\s*$|$)/i);
  let motivation = motivationMatch ? motivationMatch[1].trim() : '';
  // Remove trailing separators
  motivation = motivation.replace(/\n*---\s*$/g, '').trim();
  return motivation;
}

function extractHabitStrategy(section: string): { principle?: string; explanation?: string; example?: string; habitTip?: string } | undefined {
  const habitStrategyMatch = section.match(/###\s*\*\*Habit Strategy[\s\S]*?\*\*\s*\n([\s\S]*?)(?=###\s*\*\*(?:Content|Action|Motivation|Follow-Up|Intent|Keywords)|---\s*$|$)/i);
  
  if (!habitStrategyMatch) return undefined;
  
  const strategyText = habitStrategyMatch[1];
  const principleMatch = strategyText.match(/\*\*Strategy:\*\*|\*\*Principle:\*\*\s*([^\r\n]+)/);
  const explanationMatch = strategyText.match(/\*\*Explanation:\*\*\s*([^\r\n]+)/);
  const exampleMatch = strategyText.match(/\*\*Example:\*\*\s*([^\r\n]+)/);
  const tipMatch = strategyText.match(/\*\*Habit Tip:\*\*|\*\*Tip:\*\*\s*([^\r\n]+)/);
  
  return {
    principle: principleMatch?.[1]?.trim(),
    explanation: explanationMatch?.[1]?.trim(),
    example: exampleMatch?.[1]?.trim(),
    habitTip: tipMatch?.[1]?.trim(),
  };
}

function extractFollowUpQuestion(section: string): string {
  const followUpMatch = section.match(/###\s*\*\*Follow-Up (?:Question|Questions)\*\*\s*\n([\s\S]*?)(?=###\s*\*\*(?:Content|Action|Motivation|Habit|Intent|Keywords)|---\s*$|$)/i);
  if (!followUpMatch) return '';
  
  let followUp = followUpMatch[1].trim();
  if (followUp.includes('-')) {
    const firstLine = followUp.split('\n')[0].trim();
    followUp = firstLine.replace(/^-\s*/, '');
  }
  return followUp;
}

function parseMarkdownIntentPatterns(section: string): string[] {
  const patterns: string[] = [];
  const intentPatternsMatch = section.match(/###\s*\*\*Intent Patterns?\*\*\s*\n([\s\S]*?)(?=###|---\s*$|$)/i);
  
  if (!intentPatternsMatch) return patterns;
  
  const patternsText = intentPatternsMatch[1];
  const lines = patternsText.split('\n');
  
  for (const line of lines) {
    const trimmed = line.trim();
    if (!trimmed) continue;
    if (trimmed.match(/^(PRIMARY|SECONDARY|TIER\s+\d+|🎯)/i)) continue;
    if (trimmed.match(/^The\s+["'].*?["']\s+cluster:/i)) continue;
    if (trimmed.match(/^(QUESTIONS|CLUSTER|INTENTS?):/i) && !trimmed.match(/^[-•]/)) continue;
    
    if (trimmed.match(/^[-•]\s+/)) {
      let pattern = trimmed.replace(/^[-•]\s+/, '').trim();
      pattern = pattern.replace(/^["']|["']$/g, '').trim();
      pattern = pattern.replace(/\s*\[\+.*?\]\s*$/, '').trim();
      pattern = pattern.replace(/\s*\([^)]*\)\s*$/, '').trim();
      if (pattern && pattern.length >= 3) {
        patterns.push(pattern);
      }
    }
  }
  
  return patterns;
}

function parseMarkdownKeywords(section: string): { [category: string]: string[] } {
  const keywords: { [category: string]: string[] } = {};
  const keywordsMatch = section.match(/###\s*\*\*Keywords?\*\*\s*\n([\s\S]*?)(?=###|---\s*$|$)/);
  
  if (!keywordsMatch) return keywords;
  
  const keywordsText = keywordsMatch[1];
  const lines = keywordsText.split('\n');
  let currentCategory = 'Everyday Language';
  
  for (const line of lines) {
    const trimmed = line.trim();
    if (!trimmed) continue;
    
    // Check if it's a category header
    if (trimmed.match(/^(⚡|💪|🔬|🌡️|🎭|❓|CRITICAL|PRIMARY|SECONDARY|SCIENTIFIC|SYMPTOM|EMOTIONAL|QUESTION)/) ||
        trimmed.match(/^[A-Z][A-Z\s&/]+KEYWORDS?/)) {
      currentCategory = trimmed
        .replace(/^(⚡|💪|🔬|🌡️|🎭|❓)\s*/, '')
        .replace(/\s*KEYWORDS?.*$/i, '')
        .replace(/\([^)]*\)/g, '')
        .trim();
      if (!currentCategory || currentCategory.length < 3) currentCategory = 'Everyday Language';
      keywords[currentCategory] = [];
      continue;
    }
    
    if (trimmed.startsWith('-') || trimmed.startsWith('•')) {
      let keyword = trimmed.replace(/^[-•]\s+/, '').trim();
      keyword = keyword.replace(/^["']|["']$/g, '').trim();
      if (keyword && keyword.length >= 2) {
        if (!keywords[currentCategory]) {
          keywords[currentCategory] = [];
        }
        keywords[currentCategory].push(keyword);
      }
    }
  }
  
  return keywords;
}

function escapeYAMLString(str: string): string {
  return str.replace(/\\/g, '\\\\').replace(/"/g, '\\"').replace(/\n/g, '\\n');
}

function convertToYAML(metadata: { persona: string; topic: string; subtopic: string }, content: string, actionTips: string[], motivationNudge: string, habitStrategy: any, followUpQuestion: string, intentPatterns: string[], keywords: { [category: string]: string[] }): string {
  let yaml = '---\n';
  yaml += `persona: "${metadata.persona}"\n`;
  yaml += `topic: "${metadata.topic}"\n`;
  yaml += `subtopic: "${metadata.subtopic}"\n`;
  yaml += '\n';
  
  // Content
  yaml += 'content_text: |\n';
  const contentLines = content.split('\n');
  for (const line of contentLines) {
    yaml += `  ${line}\n`;
  }
  yaml += '\n';
  
  // Action tips
  if (actionTips.length > 0) {
    yaml += 'action_tips:\n';
    for (const tip of actionTips) {
      yaml += `  - "${escapeYAMLString(tip)}"\n`;
    }
    yaml += '\n';
  }
  
  // Motivation nudge
  if (motivationNudge) {
    yaml += `motivation_nudge: "${escapeYAMLString(motivationNudge)}"\n`;
    yaml += '\n';
  }
  
  // Habit strategy
  if (habitStrategy && (habitStrategy.principle || habitStrategy.explanation || habitStrategy.example || habitStrategy.habitTip)) {
    yaml += 'habit_strategy:\n';
    if (habitStrategy.principle) {
      yaml += `  principle: "${escapeYAMLString(habitStrategy.principle)}"\n`;
    }
    if (habitStrategy.explanation) {
      yaml += `  explanation: "${escapeYAMLString(habitStrategy.explanation)}"\n`;
    }
    if (habitStrategy.example) {
      yaml += `  example: "${escapeYAMLString(habitStrategy.example)}"\n`;
    }
    if (habitStrategy.habitTip) {
      yaml += `  habit_tip: "${escapeYAMLString(habitStrategy.habitTip)}"\n`;
    }
    yaml += '\n';
  }
  
  // Follow-up question
  if (followUpQuestion) {
    yaml += `follow_up_question: "${escapeYAMLString(followUpQuestion)}"\n`;
    yaml += '\n';
  }
  
  // Intent patterns
  if (intentPatterns.length > 0) {
    yaml += 'intent_patterns:\n';
    for (const pattern of intentPatterns) {
      yaml += `  - "${escapeYAMLString(pattern)}"\n`;
    }
    yaml += '\n';
  }
  
  // Keywords
  if (Object.keys(keywords).length > 0) {
    yaml += 'keywords:\n';
    for (const [category, keywordList] of Object.entries(keywords)) {
      if (keywordList.length > 0) {
        yaml += `  ${category}:\n`;
        for (const keyword of keywordList) {
          yaml += `    - "${escapeYAMLString(keyword)}"\n`;
        }
      }
    }
  }
  
  yaml += '---\n';
  
  return yaml;
}

function convertFile(inputPath: string, outputPath: string) {
  console.log(`Converting ${path.basename(inputPath)}...`);
  
  const content = fs.readFileSync(inputPath, 'utf-8');
  const sections = parseMarkdownSections(content);
  
  let yamlOutput = '';
  let convertedCount = 0;
  
  for (const sectionText of sections) {
    const metadata = extractMarkdownMetadata(sectionText);
    if (!metadata) continue;
    
    const contentText = extractContentText(sectionText);
    const actionTips = extractActionTips(sectionText);
    const motivationNudge = extractMotivationNudge(sectionText);
    const habitStrategy = extractHabitStrategy(sectionText);
    const followUpQuestion = extractFollowUpQuestion(sectionText);
    const intentPatterns = parseMarkdownIntentPatterns(sectionText);
    const keywords = parseMarkdownKeywords(sectionText);
    
    yamlOutput += convertToYAML(metadata, contentText, actionTips, motivationNudge, habitStrategy, followUpQuestion, intentPatterns, keywords);
    yamlOutput += '\n';
    convertedCount++;
  }
  
  fs.writeFileSync(outputPath, yamlOutput, 'utf-8');
  console.log(`✓ Converted ${convertedCount} sections`);
}

// Main execution
const knowledgeBaseDir = path.join(process.cwd(), 'knowledge-base');

// Convert files
const filesToConvert = [
  'Sleep Disturbances.md',
  'Metabolic Changes & Weight Gain.md',
];

for (const filename of filesToConvert) {
  const inputPath = path.join(knowledgeBaseDir, filename);
  const outputPath = path.join(knowledgeBaseDir, filename.replace('.md', '_converted.md'));
  
  if (fs.existsSync(inputPath)) {
    convertFile(inputPath, outputPath);
  } else {
    console.log(`⚠️  File not found: ${filename}`);
  }
}

console.log('\n✓ Conversion complete! Review the _converted.md files and replace originals if satisfied.');

